pyartcd/pyartcd/pipelines/gen_assembly.py

```diff
@@ -190,10 +194,23 @@ async def _create_or_update_pull_request(self, assembly_definition: OrderedDict)
         head = f"{match[1]}:{branch}"
         base = self.group
         if self.runtime.dry_run:
-            self._logger.warning("[DRY RUN] Would have created pull-request with head '%s', base '%s' title '%s', body '%s'", head, base, title, body)
+            if self.auto_trigger_build_sync:
+                self._logger.info(
+                    "[DRY RUN] Would have triggered build-sync with the PR assembly definition")
+            self._logger.info("[DRY RUN] Would have created pull-request with head '%s', base '%s' title '%s', "
+                              "body '%s'", head, base, title, body)
             d = {"html_url": "https://github.example.com/foo/bar/pull/1234", "number": 1234}
             result = namedtuple('pull_request', d.keys())(*d.values())
             return result
+
+        if self.auto_trigger_build_sync:
+            self._logger.info("Triggering build-sync")
+            build_version = self.group.split("-")[1]  # eg: 4.14 from openshift-4.14
+            # we're not passing doozer_data_path to build-sync because we always create branch on the base repo
+            start_build_sync(build_version=build_version,
```

Comment:
at this time the pr is not created

Reply:
The PR is created at line 222. This is called too early.